lib/index.js

```diff
@@ -162,6 +164,31 @@ FSTree.prototype.calculatePatch = function(otherFSTree, isEqual) {
   return removals.reverse().concat(additions);
 };
 
+FSTree.applyPatch = function(source, output, patch) {
```

Comment:
I went ahead and updated this to accept a patch straight from `calculatePatch` and use a visitor/delegate pattern. Works nicely.